cmds/webboot/webboot.go

```diff
@@ -207,6 +207,32 @@ func (d *DirOption) exec(uiEvents <-chan ui.Event) (menu.Entry, error) {
 	return menu.PromptMenuEntry("Distros", "Choose an option:", entries, uiEvents)
 }
 
+func mirrorMenu(entry menu.Entry, uiEvents <-chan ui.Event, link string) (url string, mirrorNameForTestPurposes string, err error) {
+	// Code for after the specific distro has been selected.
+	// Looks up the distro.
+	distro := supportedDistros[entry.Label()]
+	if len(distro.mirrors) > 0 {
+		// Make an array of type menu.Entry to store the mirrors of the
+		// particular distro selected. Then, display the mirror options.
+		entries := make([]menu.Entry, len(distro.mirrors))
+		for i := range entries {
+			entries[i] = &distro.mirrors[i]
+		}
+		entry, err = menu.PromptMenuEntry("Available Mirrors", "Choose an option:", entries, uiEvents)
+		if err != nil {
+			return "", "", err
+		}
+	}
+	// Iterate through the mirrors of the distro to select the appropriate link.
+	i := 0
+	for i := range distro.mirrors {
+		if distro.mirrors[i].name == entry.Label() {
+			link = distro.mirrors[i].url
+		}
+	}
```

Comment:
Not a big deal since we don't have very many mirrors, but consider stopping when you find the mirror you're looking for instead of looping through all of distro.mirrors.